tests/components/blebox/test_config_flow.py

```diff
@@ -0,0 +1,197 @@
+        result = await flow.async_step_user()
+        result = await flow.async_step_user({"host": "172.2.3.4", "port": 80})
+        assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_flow_with_unknown_failure(hass, product_class_mock):
+    """Test that config flow works."""
+    with product_class_mock as products_class:
+        products_class.async_from_host = CoroutineMock(side_effect=RuntimeError)
+
+        flow = init_config_flow(hass)
+        result = await flow.async_step_user()
+        result = await flow.async_step_user({"host": "172.2.3.4", "port": 80})
+        assert result["errors"] == {"base": "unknown"}
+
+
+async def test_flow_with_unsupported_version(hass, product_class_mock):
+    """Test that config flow works."""
+    with product_class_mock as products_class:
+        products_class.async_from_host = CoroutineMock(
+            side_effect=blebox_uniapi.error.UnsupportedBoxVersion
+        )
+
+        flow = init_config_flow(hass)
+        result = await flow.async_step_user()
+        result = await flow.async_step_user({"host": "172.2.3.4", "port": 80})
+        assert result["errors"] == {"base": "unsupported_version"}
+
+
+async def test_already_configured(hass):
+    """Test that same device cannot be added twice."""
+
+    feature = mock_only_feature(
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-gateBox-1afe34db9437-0.position",
+        full_name="gateBox-0.position",
+        device_class="gate",
+        state=0,
+        async_update=CoroutineMock(),
+        current=None,
+    )
+
+    setup_product_mock(
+        "covers", [feature], "homeassistant.components.blebox.Products",
+    )
+
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My gate controller")
+    type(product).model = PropertyMock(return_value="gateController")
+    type(product).type = PropertyMock(return_value="gateBox")
+    type(product).brand = PropertyMock(return_value="BleBox")
+    type(product).firmware_version = PropertyMock(return_value="1.23")
+    type(product).unique_id = PropertyMock(return_value="abcd0123ef5678")
+
+    config = mock_config("172.1.2.3")
+    config.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(config.entry_id)
+    await hass.async_block_till_done()
+
+    result = await hass.config_entries.flow.async_init(
+        "blebox", context={"source": "user"}, data={"host": "172.1.2.3", "port": 80},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "address_already_configured"
+
+
+async def test_async_setup(hass):
+    """Test async_setup (for coverage)."""
+    assert await async_setup_component(hass, "blebox", {"host": "172.2.3.4"})
+
+
+async def test_async_setup_entry(hass):
+    """Test async_setup_entry (for coverage)."""
+    config = mock_config()
+    config.add_to_hass(hass)
+
+    with patch.object(
+        hass.config_entries,
```

Comment:
Please don't patch home assistant internals as they may change.

Reply:
You could do something like 
```
from tests.common import MockConfigEntry, load_fixture
...
    with patch("some_part_of_blebox_pypi_package", side_effect=_handle_mock_api_request):
        entry = MockConfigEntry(
            domain=DOMAIN, data={CONF_USERNAME: "mock", CONF_PASSWORD: "mock"}
        )
        entry.add_to_hass(hass)
        await hass.config_entries.async_setup(entry.entry_id)
        await hass.async_block_till_done()

  .. check that something you expect to be there is ...
```

Reply:
Fixed in https://github.com/home-assistant/core/pull/32664/commits/0af15592cea1ca8b92f3f89631c34f0711cd0d02

(At first I didn't realize you meant to use higher level tests, which do make a lot more sense).

As for grouping tests together, I think config_entries setup/remove tests shouldn't be in `test_config_flow.py` (which is basically just for user flow in my case). They should probably be repeated for every platform I guess. But that may create a lot more repetitive tests that could slow down CI builds noticeably. If you have suggestions, let me know, thanks!